lib/sth_database.js

```diff
@@ -1841,6 +1841,490 @@ function getNotificationInfo(data, callback) {
+      entityId = data.entityId,
+      entityType = data.entityType,
+      attrName = data.attrName;
+  var isAggregated = options.isAggregated;
+
+  getCollection(
+    {
+      service: service,
+      collection: sthConfig.COLLECTION_PREFIX + 'collection_names'
+    },
+    {
+      isAggregated: false,
+      shouldCreate: false,
+      shouldStoreHash: false,
+      shouldTruncate: false
+    },
+    function (err, hashCollection) {
+      if (err) {
+        return process.nextTick(callback.bind(null, err));
+      }
+      hashCollection.findOne(
+        {
+          _id: collectionName
+        },
+        function(err, result) {
+          if (err) {
+            return process.nextTick(callback.bind(null, err));
+          }
+          if (result && result.isAggregated === isAggregated &&
+            result.service === service && result.servicePath === servicePath &&
+            (entityId ? result.entityId === entityId : true) &&
+            (entityType ? result.entityType === entityType : true) &&
+            (attrName ? result.attrName === attrName : true)) {
+            db.db(getDatabase(service)).dropCollection(
+              collectionName,
+              function(err) {
+                if (err) {
+                  return process.nextTick(callback.bind(null, err));
+                }
+                hashCollection.remove(
+                  {
+                    _id: collectionName
+                  },
+                  {
+                    writeConcern: {
+                      w: !isNaN(sthConfig.WRITE_CONCERN) ?
+                        parseInt(sthConfig.WRITE_CONCERN, 10) : sthConfig.WRITE_CONCERN
+                    }
+                  },
+                  callback
+                );
+              }
+            );
+          } else {
+            process.nextTick(callback);
+          }
+        }
+      );
+    }
+  );
+}
+
+/**
+ * Removes the data associated to the attributes specified in the provided data in case of collection per
+ *  service path data model
+ * @param {object} data It is an object including the following properties:
+ *  - {string} service The service
+ *  - {string} servicePath The service path
+ *  - {string} entityId The entity id
+ *  - {string} entityType The entity type
+ *  - {string} attrName The attribute name. In case no attribute name is provided, all the attributes of the
+ *    provided entity are removed
+ * @param {object} options It is an object including the following properties:
+ *  - {boolean} isAggregated Flag indicating if the operation refers to aggregated data. Raw data otherwise
+ * @param callback The callback to call with error or the result of the operation
+ */
+function removeDataIfCollection4ServicePath(data, options, callback) {
+  var service = data.service,
+      servicePath = data.servicePath,
+      entityId = data.entityId,
+      entityType = data.entityType,
+      attrName = data.attrName;
+  var isAggregated = options.isAggregated;
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-sth-comet/commit/7cf94a6c7d3ea3a1a0f7f869ed227f63d43f9dd1